servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublishAndSubscribeOnCompletables.java

```diff
@@ -71,17 +86,20 @@ static Completable subscribeOn(final Completable original,
         @Override
         void handleSubscribe(final Subscriber subscriber,
                              final AsyncContextMap contextMap, final AsyncContextProvider contextProvider) {
+            final Subscriber upstreamSubscriber;
             try {
                 BooleanSupplier shouldOffload = shouldOffload();
-                Subscriber upstreamSubscriber =
+                upstreamSubscriber =
                         new CompletableSubscriberOffloadedTerminals(subscriber, shouldOffload, executor());
 
                 // Note that the Executor is wrapped by default to preserve AsyncContext, so we don't have to re-wrap
```

Comment:
This comment was linked to the line that is now moved further down, consider moving it too.

Reply:
It is actually a comment at the site where the (unnecessary, per the comment) async context wrapping would be done. I believe it is in the correct location since if the wrapping was needed it would be in that try/catch block.